enclave/src/lib.rs

```diff
@@ -425,6 +425,69 @@ pub unsafe extern "C" fn accept_pdex_accounts(
     sgx_status_t::SGX_SUCCESS
 }
 
+#[no_mangle]
+pub unsafe extern "C" fn send_disk_data(encoded_data: *const u8, data_size: usize) -> sgx_status_t {
+    let mut data = slice::from_raw_parts(encoded_data, data_size);
+
+    let decoded: polkadex_sgx_primitives::StorageData =
+        if let Ok(data) = polkadex_sgx_primitives::StorageData::decode(&mut data) {
+            data
+        } else {
+            return sgx_status_t::SGX_ERROR_UNEXPECTED;
+        };
+
+    let balances_data: Vec<(Vec<u8>, polkadex_balance_storage::Balances)> = decoded
+        .balances
+        .into_iter()
+        .map(|entry| {
+            (
+                polkadex_balance_storage::PolkadexBalanceKey::from(
+                    entry.asset_id,
+                    entry.account_id,
+                )
+                .encode(),
+                polkadex_balance_storage::Balances::from(entry.free, entry.reserved),
+            )
+        })
+        .collect();
+    let nonce_data: Vec<(Vec<u8>, u32)> = decoded
```

Comment:
Insteda of `Vec<(Vec<u8>, u32)>`, this should be `Vec<NonceData>`,  or is there a difference?

Reply:
Changed.